### DISCUSSION TITLE
`PATCH` request and nullable fields

### DISCUSSION THREAD

**Discussion by leon0399** [spatie/laravel-data#855] [Q&A]
How do y'all handle your `PATCH` requests with nullable fields?

What I mean:

In `PATCH` conventionally, these requests should behave differently:

1. Should set both `title` and `artist` to null.
  ```json
  {
    "data": {
      "title": null, 
      "artist": null
    }
  }
  ```

2. Should set `artist` to null and leave `title` as-is.
  ```json
  {
    "data": {
      "artist": null,
    }
  }
  ```
  
With standard Laravel FormRequests, it is pretty easy: just do `$model->update($validated)`

But with DTOs this wouldn't work:

```php
class SongData extends Data
{
    public function __construct(
        public ?string $title,
        public ?string $artist,
    ) {
    }
    
    public static function rules(): array
    {
        return [
            'title' => ['nullable', 'string'],
            'artist' => ['nullable', 'string'],
        ];
    }
}

dd($songData->title); // is it null because it explicitly requested as null, or because it wasn't in request?
```

**Comment by stfndamjanovic** [comment]
Did you try passing the model as a first parameter and the request as a second  while creating data object?
```php
SongData::from($song, $request);
```

Then, the package will merge the missing fields from the request. 

This is maybe a workaround, but it should work.

I don't know if maintainers can propose us a better solution to this problem?
